src/api/v1/db/oracledb/pizzas-dao.js

```diff
@@ -0,0 +1,93 @@
+import _ from 'lodash';
+
+import { getConnection } from 'api/v1/db/oracledb/connection';
+import { getDoughById } from 'api/v1/db/oracledb/doughs-dao';
+import { serializePizza } from 'api/v1/serializers/pizzas-serializer';
+
+const pizzaColumns = {
+  id: 'ID',
+  doughId: 'DOUGH_ID',
+  name: 'NAME',
+  bakeTime: 'BAKE_TIME',
+  ovenTemp: 'OVEN_TEMP',
+  specialInstructions: 'SPECIAL_INSTRUCTIONS',
+};
+
+const pizzaColumnAliases = _.map(pizzaColumns,
+  (columnName, attributeName) => `${columnName} AS "${attributeName}"`)
+  .join(', ');
+
+/**
+ * Fetch all the ingredients associated with the pizza with id
+ *  `pizzaId`
+ *
+ * @param {string} pizzaId
+ * @returns {Array} an array of unserialized ingredient objects
+ */
+const getPizzaIngredients = async (pizzaId) => {
+  const query = 'SELECT INGREDIENTS.ID AS "id", INGREDIENTS.TYPE AS "ingredientType", INGREDIENTS.NAME as "name", INGREDIENTS.NOTES AS "notes" FROM PIZZA_INGREDIENTS INNER JOIN INGREDIENTS ON PIZZA_INGREDIENTS.INGREDIENT_ID = INGREDIENTS.ID WHERE PIZZA_INGREDIENTS.PIZZA_ID = :id';
+  const bindParams = { id: pizzaId };
+  const connection = await getConnection();
+  try {
+    const result = await connection.execute(
+      query,
+      bindParams,
+    );
+    return result.rows;
+  } finally {
+    connection.close();
+  }
+};
+
+/**
+ * Get a "deserialized" dough object using `getDoughById`
+ *
+ * @param {string} doughId the ID of the dough to get
+ * @returns {object} a raw dough object
+ */
+const getPizzaDough = async (doughId) => {
+  const serializedDough = await getDoughById(doughId);
+
+  if (serializedDough === null) return {};
+
+  const deserializedDough = {
```

Comment:
Seems like a waste to serialize the dough just to unserialize it. I believe there was talk about moving serialization into the daos but seeing this makes me question that. Seems less modular. Thoughts?

Reply:
I definitely think that having the path handler do serialization instead of the DAO makes sense in this case. This was the simplest option given the code base as it exists right now, but I'm totally in agreement that it's not an ideal solution.